src/nat/runtime/session.py

```
@@ -609,6 +671,28 @@ def set_metadata_from_http_request(self, request: Request) -> None:
         if workflow_run_id:
             self._context_state.workflow_run_id.set(workflow_run_id)
 
+        # NAT observability headers (do not overwrite if already set)
+        nat_trace_id = request.headers.get("x-nat-trace-id")
+        if nat_trace_id and not self._context_state.workflow_trace_id.get():
+            try:
+                self._context_state.workflow_trace_id.set(int(nat_trace_id, 16))
+            except Exception:
+                pass
```

Ruff (0.14.11)

`try`-`except`-`pass` detected, consider logging the exception

(S110)